internal/server/event_server.go

```diff
@@ -124,10 +129,16 @@ func (s *EventServer) cleanupMetadataMiddleware(h http.Handler) http.Handler {
 
 		cleanupMetadata(event)
 
-		ctxWithEvent := context.WithValue(r.Context(), eventContextKey{}, event)
-		reqWithEvent := r.WithContext(ctxWithEvent)
+		logger := s.logger.WithValues(
+			"reconciler kind", event.InvolvedObject.Kind,
+			"name", event.InvolvedObject.Name,
+			"namespace", event.InvolvedObject.Namespace)
```

Comment:
Using `FromContext()` and `IntoContext()` does not help testing that `eventMiddleware()` will correctly inject the objects into the request context, because they use structs. I'm gonna submit a patch now that is an attempt of making struct-agnostic logging interfaces so that we can test the behavior of `eventMiddleware()`. If that's too much/too complicated, we can just use `FromContext()` and `IntoContext()` and not test the behavior of this middleware instead.

Reply:
Agreed.

Reply:
Done, thanks!